daemon/cmd/app.go

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/cilium/cilium/api/v1/server"
+)
+
+const (
+	startTimeout = 5 * time.Minute
+	stopTimeout  = 5 * time.Minute
+)
+
+// tempDaemonModule wraps the existing daemon initialization
```

Comment:
nit:
```suggestion
// daemonModule wraps the existing daemon initialization
```

Even if we intend for this to be temporary, the reality is that after this PR is merged, there is a daemon module. We can argue about whether it should be there or not, but until we make enough changes to actually remove the need for an overall daemon module, it's going to stay here. And as always, even with the best intentions, code you think is temporary will likely live longer than you expect.

Reply:
Good point. I'm going to go with the suggestion you had that `runDaemon` returns an fx.Option, which removes the need for this and will leave less stuff that sticks around :)